### ISSUE TITLE
holdDelay

### ISSUE THREAD

**Issue by martienvlent** [NorthwoodsSoftware/GoJS#24]
If I time the holdDelay in the Minimal Sample, like so:
```
	myDiagram.toolManager.doMouseDown=function(){
		start = new Date().getTime();
		go.Tool.prototype.doMouseDown.call(myDiagram.toolManager)
	}

	myDiagram.toolManager.dragSelectingTool.doActivate=function(){
		console.log('Execution time: ' + (new Date().getTime() - start)); 
		go.DragSelectingTool.prototype.doActivate.call(myDiagram.toolManager.dragSelectingTool)
	}
```
then the dragSelectingTool starts very much earlier then the value of holdDelay, down to a 180ms.

And if I try it on a shape with:
```
	mouseHold:function(e,p){
		console.log('Execution time: ' + (new Date().getTime() - start)); 
	}
```
then the result varies mostly between about 200ms and 800ms, but sometimes it's 50ms and at other times it does not fire at all.

Furthermore, the value of the actual holdDelay does not have an effect on the results.

Please advice.

**Comment by WalterNorthwoods** [comment]
The **ToolManager.holdDelay** property just controls when the **Tool.doWaitAfter** method is called.  The default value is 850 milliseconds.  A timer is started upon mouse-down, is restarted on mouse-move beyond a certain distance (**Tool.isBeyondDragSize**), and is cancelled upon mouse-up and in a few other circumstances.

The **DragSelectingTool.delay** property controls how long to wait before **DragSelectingTool.canStart** can return true.  The default value is 175 milliseconds.  No timer is involved.

So the **ToolManager.holdDelay** property should not affect the behavior of the **DragSelectingTool**.

I believe the user does have to move the mouse after a mouseHold event in order to get a new mouseHold event.  The same goes for mouseHover events, which implement tooltips.

I cannot explain why the timings are less than 800ms.  We'll investigate.  I suspect it's interacting with waiting for mouseHover events.

**Comment by WalterNorthwoods** [comment]
Ah, this is a bug in your test code.  You wrote:
```
    myDiagram.toolManager.doMouseDown=function(){
        start = new Date().getTime();
        go.Tool.prototype.doMouseDown.call(myDiagram.toolManager)
    }
```

But the call to the overridden method should have been:
```
    myDiagram.toolManager.doMouseDown=function(){
        start = new Date().getTime();
        go.ToolManager.prototype.doMouseDown.call(myDiagram.toolManager)
    }
```

Basically, your code was not calling the normal **ToolManager.doMouseDown** functionality.

With the fix I am consistently getting times of about 850-860 ms.

**Comment by martienvlent** [comment]
Oops.
Thanks for the swift reply.

Groeten, Martien
________________________________
Van: WalterNorthwoods<mailto:notifications@github.com>
Verzonden: ‎16-‎6-‎2016 14:48
Aan: NorthwoodsSoftware/GoJS<mailto:GoJS@noreply.github.com>
CC: Martien van Lent<mailto:mvlent@outlook.com>; Author<mailto:author@noreply.github.com>
Onderwerp: Re: [NorthwoodsSoftware/GoJS] holdDelay (#24)

Ah, this is a bug in your test code.  You wrote:
```
    myDiagram.toolManager.doMouseDown=function(){
        start = new Date().getTime();
        go.Tool.prototype.doMouseDown.call(myDiagram.toolManager)
    }
```

But the call to the overridden method should have been:
```
    myDiagram.toolManager.doMouseDown=function(){
        start = new Date().getTime();
        go.ToolManager.prototype.doMouseDown.call(myDiagram.toolManager)
    }
```

Basically, your code was not calling the normal **ToolManager.doMouseDown** functionality.

With the fix I am consistently getting times of about 850-860 ms.

---
